File: CrazyBooks_Final/CrazyBooks.Web/wwwroot/Scripts/Models/Room.js
class Room extends Entity
{
    constructor(json)
    {
        super(json);

        if (json) {
            this.Code = json.code;
            this.Color = json.color;
            this.Capacity = json.capacity;
            this.Accessible = json.accessible;
        }
        else {
            this.Code = "";
            this.Color = "";
            this.Capacity = 0;
            this.Accessible = "";
        }
    }
}

File: CrazyBooks_Final/CrazyBooks.Lib/Models/Room.cs
using Newtonsoft.Json;
using System;
using System.Collections.Generic;

namespace CrazyBooks.Lib.Models
{
    public class Room : Entity
    {
        public string Code { get; set; }
        public string Color { get; set; }
        public string Capacity { get; set; }
        public int Accessible { get; set; }
        [JsonIgnore]
        public ICollection<RoomReservation> RoomReservations { get; set; }

    }
}
